internal/controllers/core/liveupdate/reconciler.go

```diff
@@ -925,63 +936,77 @@ func (r *Reconciler) applyInternal(
+				ContainerName:      cInfo.ContainerName.String(),
+				ContainerID:        cInfo.ContainerID.String(),
+				PodName:            cInfo.PodID.String(),
+				Namespace:          cInfo.Namespace.String(),
+				LastFileTimeSynced: lastFileTimeSynced,
+			}
 
-		if err != nil {
-			if runFail, ok := build.MaybeRunStepFailure(err); ok {
-				// Keep running updates -- we want all containers to have the same files on them
-				// even if the Runs don't succeed
-				logger.Get(ctx).Infof("  → Failed to update container %s: run step %q failed with exit code: %d",
-					cInfo.DisplayName(), runFail.Cmd.String(), runFail.ExitCode)
-				cStatus.LastExecError = err.Error()
-				lastExecErrorStatus = &cStatus
-			} else {
-				// Something went wrong with this update and it's NOT the user's fault--
-				// likely a infrastructure error. Bail, and fall back to full build.
-				result.Failed = &v1alpha1.LiveUpdateStateFailed{
-					Reason:  "UpdateFailed",
-					Message: fmt.Sprintf("Updating pod %s: %v", cStatus.PodName, err),
+			if err != nil {
+				if runFail, ok := build.MaybeRunStepFailure(err); ok {
+					// Keep running updates -- we want all containers to have the same files on them
+					// even if the Runs don't succeed
+					logger.Get(ctx).Infof("  → Failed to update container %s: run step %q failed with exit code: %d",
+						cInfo.DisplayName(), runFail.Cmd.String(), runFail.ExitCode)
+					cStatus.LastExecError = err.Error()
+				} else {
+					// Something went wrong with this update, and it's NOT the user's fault--
+					// likely an infrastructure error. Bail, and fall back to full build.
+					failed = &v1alpha1.LiveUpdateStateFailed{
+						Reason:  "UpdateFailed",
+						Message: fmt.Sprintf("Updating pod %s: %v", cStatus.PodName, err),
+					}
 				}
-				return result
+			} else {
+				logger.Get(ctx).Infof("  → Container %s updated!", cInfo.DisplayName())
 			}
-		} else {
-			logger.Get(ctx).Infof("  → Container %s updated!", cInfo.DisplayName())
-			if lastExecErrorStatus != nil {
-				// This build succeeded, but previously at least one failed due to user error.
-				// We may have inconsistent state--bail, and fall back to full build.
-				result.Failed = &v1alpha1.LiveUpdateStateFailed{
-					Reason: "PodsInconsistent",
-					Message: fmt.Sprintf("Pods in inconsistent state. Success: pod %s. Failure: pod %s. Error: %v",
-						cStatus.PodName, lastExecErrorStatus.PodName, lastExecErrorStatus.LastExecError),
-				}
-				return result
+			results <- &v1alpha1.LiveUpdateStatus{
+				Containers: []v1alpha1.LiveUpdateContainerStatus{
+					cStatus,
+				},
+				Failed: failed,
 			}
-		}
+			return nil
+		})
+	}
 
-		result.Containers = append(result.Containers, cStatus)
+	for {
+		select {
+		case res := <-results:
+			if res.Failed != nil {
+				cancel()
+				result.Failed = res.Failed
+			}
+			result.Containers = append(result.Containers, res.Containers...)
+		case <-gErrorC(g):
+		case <-ctx.Done():
+			cancel()
```

Comment:
this is going to create a memory leak i think ... the channel send will block forever if the outer function returns early without returning all values.

you'll need to drain the channel here.

though honestly, i think this would be a lot simpler if you used a slice to accumulate the results, and didn't use a channel at all (as the errgroup example code does)